tools/rostopic/src/rostopic/__init__.py

```diff
@@ -1933,6 +1933,15 @@ def stdin_publish(pub, msg_class, rate, once, filename, verbose):
     else:
         iterator = stdin_yaml_arg
 
+    # determine whether there is exactly one message
+    exactly_one_message = True
```

Comment:
Shouldn't this value start with `False`? If the iterator has no elements the value of the flag incorrectly reports `True`.